core/state_processor.go

```diff
@@ -79,41 +79,79 @@ func (p *StateProcessor) Process(
 		allLogs  []*types.Log
 		gp       = new(GasPool).AddGas(block.GasLimit())
 	)
-	beneficiary, err := p.bc.GetECDSAFromCoinbase(header)
 
+	beneficiary, err := p.bc.GetECDSAFromCoinbase(header)
 	if err != nil {
 		return nil, nil, nil, 0, nil, err
 	}
 
-	// Iterate over and process the individual transactions
-	for i, tx := range block.Transactions() {
```

Comment:
need a forking condition on pre-staking epoch, have to use the old logic before that.

Reply:
staking transactions won't be included in the block due to prestaking check in the `CommitTransactions` right?